codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/HttpBoundProtocolPayloadGenerator.kt

```diff
@@ -53,11 +54,24 @@ data class EventStreamBodyParams(
     val additionalPayloadContext: AdditionalPayloadContext,
 )
 
+data class StreamPayloadSerializerParams(
+    val symbolProvider: SymbolProvider,
+    val runtimeConfig: RuntimeConfig,
+    val member: MemberShape,
+    val payloadName: String?,
+)
+
+data class StreamPayloadSerializer(
+    val outputT: (RustWriter, StreamPayloadSerializerParams) -> Unit,
+    val renderPayload: (RustWriter, StreamPayloadSerializerParams) -> Unit,
+)
```

Comment:
Seems like this should be an interface rather than a data class.

Reply:
Now that I'm looking at it, yes totally. Addressed in ab57dce.